scripts/tools/metricProgrammable.yml

```diff
@@ -0,0 +1,513 @@
+type: macro
+desc: "Maximum metric programmable name string size"
+version: "1.8"
+name: $T_MAX_METRIC_PROGRAMMABLE_NAME_EXP
+value: "128"
+--- #--------------------------------------------------------------------------
+type: macro
+desc: "Maximum metric programmable description string size"
+version: "1.8"
+name: $T_MAX_METRIC_PROGRAMMABLE_DESCRIPTION_EXP
+value: "128"
+--- #--------------------------------------------------------------------------
+type: macro
+desc: "Maximum metric programmable component string size"
+version: "1.8"
+name: $T_MAX_METRIC_PROGRAMMABLE_COMPONENT_EXP
+value: "128"
+--- #--------------------------------------------------------------------------
+type: macro
+desc: "Maximum metric programmable parameter string size"
+version: "1.8"
+name: $T_MAX_METRIC_PROGRAMMABLE_PARAMETER_NAME_EXP
+value: "128"
+--- #--------------------------------------------------------------------------
+type: macro
+desc: "Maximum value information string size"
+version: "1.8"
+name: $T_MAX_VALUE_INFO_CSTRING_EXP
+value: "128"
+--- #--------------------------------------------------------------------------
+type: handle
+desc: "Handle of metric programmable's object"
+version: "1.8"
+class: $tMetricProgrammable
+name: "$t_metric_programmable_exp_handle_t"
+--- #--------------------------------------------------------------------------
+type: struct
+desc: "Metric Programmable properties queried using $tMetricProgrammableGetPropertiesExp"
+version: "1.8"
+class: $tMetricProgrammable
+name: $t_metric_programmable_exp_properties_t
+base: $t_base_properties_t
+members:
+    - type: char
+      name: "name[$T_MAX_METRIC_PROGRAMMABLE_NAME_EXP]"
+      desc: "[out] metric programmable name"
+    - type: char
+      name: "description[$T_MAX_METRIC_PROGRAMMABLE_DESCRIPTION_EXP]"
+      desc: "[out] metric programmable description"
+    - type: char
+      name: "component[$T_MAX_METRIC_PROGRAMMABLE_COMPONENT_EXP]"
+      desc: "[out] metric programmable component"
+    - type: uint32_t
+      name: "tierNumber"
+      desc: "[out] tier number"
+    - type: uint32_t
+      name: "domain"
+      desc: "[out] metric domain number."
+    - type: uint32_t
+      name: "parameterCount"
+      desc: "[out] number of parameters in the programmable"
+    - type: "$t_metric_group_sampling_type_flags_t"
+      name: "samplingType"
+      desc: |
+            [out] metric sampling type.
+            returns a combination of $t_metric_group_sampling_type_flags_t.
+    - type: uint32_t
+      name: "sourceId"
+      desc: "[out] unique metric source identifier(within platform)to identify the HW block where the metric is collected."
+--- #--------------------------------------------------------------------------
+type: enum
+desc: "Metric Programmable Parameter types"
+version: "1.8"
+class: $tMetricProgrammable
+name: $t_metric_programmable_param_type_exp_t
+etors:
+    - name: ETOR_DISAGGREGATION
```

Comment:
Why the `ETOR_` prefix? The scripts will expand the name. For example, `DISAGGREGATION` will expand to `ZET_METRIC_PROGRAMMABLE_PARAM_TYPE_DISAGGREGATION`.

The same comment/question follows for all the etors in this list and in the "`value_info_type`" etors below.